### PR TITLE
Create items list

### PR DESCRIPTION
**Pull request by Arturnix**

### CODE REVIEW

File: src/main/java/pl/arturzgodka/controllers/ItemsController.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    public String getItemTypesNames(Model model, @PathVariable(value="itemTypes") String itemType) {
+
+        Set<String> itemTypeNames = ItemClassesAndNamesLists.getItemTypesNames(itemType);
+        model.addAttribute("selectedItemType", itemType);
+        model.addAttribute("itemTypesNames", itemTypeNames);
+
+        return "itemTypes";
+    }
+
+    /*@RequestMapping("/Armor/")
```
Comment (blazejprzyluski) [inline src/main/java/pl/arturzgodka/controllers/ItemsController.java:39]: why the commented out code?

File: src/main/java/pl/arturzgodka/databaseutils/UserSessionFactory.java [hunk 2]
```diff
@@ -9,4 +9,9 @@ public static SessionFactory getCustomUserSessionFactory() {
         config.configure("static/hibernate.cfg.xml"); //plik w ktorym konfiguruje hibernate
         return config.buildSessionFactory();
     }
+
```
Comment (blazejprzyluski) [inline src/main/java/pl/arturzgodka/databaseutils/UserSessionFactory.java:12]: why the commented out code?

File: src/main/java/pl/arturzgodka/datamodel/ItemArmorDataModel.java [hunk 3]
```diff
@@ -25,20 +25,47 @@ public ItemArmorDataModel() {
     }
 
     public ItemArmorDataModel(List<String> itemBodyPartSlots, String id, String name, int requiredLevel, Map<String, List<String>> attributes, String armor) {
+        this.requiredLevel = requiredLevel;
+        this.armor = armor;
+    }
+ //czy te wszystkie konstruktory sa wykorzystywane???
```
Comment (blazejprzyluski) [inline src/main/java/pl/arturzgodka/datamodel/ItemArmorDataModel.java:31]: same question

File: src/main/java/pl/arturzgodka/jsonmappers/SkillMapper.java [hunk 4]
```diff
@@ -14,7 +14,7 @@
 
 public class SkillMapper {
 
-    public List<HeroSkillDataModel> mapSkillsToDataModel(String heroClass, List<String> barbarianSkills) {
+    public List<HeroSkillDataModel> mapSkillsToDataModel(String heroClass, List<String> barbarianSkills) { //change List<String> not to specific class name
```
Comment (blazejprzyluski) [inline src/main/java/pl/arturzgodka/jsonmappers/SkillMapper.java:17]: why the comment?

File: src/main/java/pl/arturzgodka/controllers/ItemsController.java (hunk 1 above)
Comment (Arturnix) [reply src/main/java/pl/arturzgodka/controllers/ItemsController.java:39]: That was first version of controller logic for displaying items list but it was divided into 2 methods. Now it is merged into 1 method to operate on both request: for displaying armor and weapon items lists.
Done, this part of commented code is out.

File: src/main/java/pl/arturzgodka/databaseutils/UserSessionFactory.java (hunk 2 above)
Comment (Arturnix) [reply src/main/java/pl/arturzgodka/databaseutils/UserSessionFactory.java:12]: Done

File: src/main/java/pl/arturzgodka/datamodel/ItemArmorDataModel.java (hunk 3 above)
Comment (Arturnix) [reply src/main/java/pl/arturzgodka/datamodel/ItemArmorDataModel.java:31]: Comments moved to //TODO section. Done

File: src/main/java/pl/arturzgodka/jsonmappers/SkillMapper.java (hunk 4 above)
Comment (Arturnix) [reply src/main/java/pl/arturzgodka/jsonmappers/SkillMapper.java:17]: Applied changes from comment. Done

**Review by blazejprzyluski — APPROVED** [review]

**Pull request merged**